File: frontend/src/services/health.ts
import { apiClient } from "./api";

export interface HealthInfo {
  ok: boolean;
  timestamp?: string;
  version?: string;
  pythonVersion?: string;
  uptimeSeconds?: number;
}

interface BackendHealthResponse {
  status: string;
  server_time?: string;
  app_version?: string;
  python_version?: string;
  uptime_seconds?: number;
  [key: string]: unknown;
}

/**
 * Health check service
 */
export const healthService = {
  /**
   * Get health status
   */
  async getHealth(): Promise<HealthInfo> {
    const response = await apiClient.get<BackendHealthResponse>("/health");
    // Map backend response format to frontend expected format
    return {
      ok: response.data.status === "ok",
      timestamp: response.data.server_time,
      version: response.data.app_version,
      pythonVersion: response.data.python_version,
      uptimeSeconds: response.data.uptime_seconds,
    };
  },
};

File: backend/server/api/schemas/health.py
"""Pydantic schemas for Health API."""

from pydantic import BaseModel, ConfigDict


class HealthResponse(BaseModel):
    """Schema for health check response."""

    model_config = ConfigDict(from_attributes=True)

    status: str
    server_time: str
    python_version: str
    system: str
    hostname: str
    process_id: int
    uptime_seconds: float | None
    app_version: str
